🤔
`* [ERROR] [QUALITY_SCALE] [test-before-setup] Integration does not raise one of {'ConfigEntryError', 'ConfigEntryNotReady', 'ConfigEntryAuthFailed'} in async_setup_entry (/home/runner/work/core/core/homeassistant/components/fronius/__init__.py)`
It does raise that in a helper function. But apparently that's not covered by the test. 
Should this be 
- `exempt` or 
- covered by some hack like
```py
try:
    helper_function()
except ConfigEntryNotReady:
    raise ConfigEntryNotReady()
``` 
- `todo` until the test supports such scenarios 
?

--- a/homeassistant/components/fronius/quality_scale.yaml
+++ b/homeassistant/components/fronius/quality_scale.yaml
@@ -11,12 +11,7 @@ rules:
     comment: |
       Single platform only, so no entity.py file.
       CoordinatorEntity is used.
-  config-flow-test-coverage:
-    status: todo
-    comment: |
-      - Remove duplicated patch of `async_setup_entry`.
-      - Assert unique_id in test_form_with_logger and test_dhcp.
-      - All tests should end in either a CREATE_ENTRY or ABORT.
+  config-flow-test-coverage: done
   config-flow: done
   dependency-transparency: done
   docs-actions:
@@ -63,12 +58,7 @@ rules:
     status: exempt
     comment: |
       This integration doesn't require authentication.
-  test-coverage:
-    status: todo
-    comment: |
-      - Use fixture `enable_all_entities` instead of custom code.
-      - Use freezer.
-      - Use snapshots.
+  test-coverage: done
   # Gold
   devices: done
   diagnostics: done

--- a/homeassistant/components/fronius/quality_scale.yaml
+++ b/homeassistant/components/fronius/quality_scale.yaml
@@ -32,10 +32,7 @@ rules:
       This integration does not subscribe to events.
   entity-unique-id: done
   has-entity-name: done
-  runtime-data:
-    status: todo
-    comment: |
-      `async_remove_config_entry_device` does not use typed ConfigEntry.
+  runtime-data: done
   test-before-configure: done
   test-before-setup:
     status: todo

--- a/homeassistant/components/fronius/quality_scale.yaml
+++ b/homeassistant/components/fronius/quality_scale.yaml
@@ -37,9 +37,12 @@ rules:
     comment: |
       `async_remove_config_entry_device` does not use typed ConfigEntry.
   test-before-configure: done
-  test-before-setup: done
+  test-before-setup:
+    status: todo
+    comment: |
+      ConfigEntryNotReady not raised in async_setup_entry but in helper function.
+      Hassfest test fails for this.
   unique-config-entry: done
-
   # Silver
   action-exceptions:
     status: exempt

